server/room-battle.ts

```diff
@@ -1338,6 +1353,364 @@ export class RoomBattle extends RoomGames.RoomGame<RoomBattlePlayer> {
+		for (const userid of [this.p1, this.p2]) {
+			buf += `<span style="text-align: right">`;
+			buf += `${this.name(userid)}: `;
+			for (let i = 0; i < this.bestOf; i++) {
+				if (this.games[i]?.winner === userid) {
+					buf += `<i class="fa fa-circle"></i>`;
+				} else {
+					buf += `<i class="fa fa-circle-o"></i>`;
+				}
+				if ((i + 1) === this.winThreshold) {
+					buf += ` | `;
+				} else {
+					buf += ` `;
+				}
+			}
+			buf += `</span><br />`;
+		}
+		buf += `<br /><br />`;
+		buf += `<table><tr>`;
+
+		for (const userid of [this.p1, null, this.p2]) {
+			if (userid === null) {
+				buf += `<td></td>`;
+				continue;
+			}
+			buf += `<td><center><strong>${this.name(userid)}</strong></center></td>`;
+		}
+
+		buf += `</tr><tr>`;
+
+		for (const userid of [this.p1, null, this.p2]) {
+			if (userid === null) {
+				buf += `<td></td>`;
+				continue;
+			}
+			const name = Users.get(userid)?.avatar || 'unknownf';
+			const url = Chat.plugins.avatars?.Avatars.src(name) || `https://${Config.routes.client}/sprites/trainers/${name}.png`;
+			buf += `<td><center>`;
+			buf += `<img src="${url}" width="80" height="80" />`;
+			buf += `</center></td>`;
+		}
+
+		buf += `</tr><tr>`;
+
+		for (const [i, slot] of ['p1', null, 'p2'].entries()) {
+			if (slot === null) {
+				buf += `<td> vs </td>`;
+				continue;
+			}
+			const team = Teams.unpack(this.options[slot as 'p1' | 'p2']?.team || "");
+			if (!team) {
+				buf += `<td>`;
+				buf += `<psicon pokemon="unknown" /> `.repeat(3);
+				buf += `<br />`;
+				buf += `<psicon pokemon="unknown" /> `.repeat(3);
+				buf += `</td>`;
+				continue;
+			}
+			const style = !i ? ' style="transform: scaleX(-1)"' : "";
+			buf += `<td>`;
+			for (const [j, set] of team.entries()) {
+				if (j % 3 === 0 && j > 1) buf += `<br />`;
+				buf += `<psicon pokemon="${set.species}"${style}/>`;
+			}
+			buf += `</td>`;
+		}
+		buf += `</tr></table>`;
+
+		this.room.add(`|fieldhtml|<center>${buf}</center>`);
+		buf = this.games.map(({battle, winner}, index) => {
+			let progress = `in progress`;
+			if (winner) progress = `winner: ${this.name(winner)}`;
+			if (winner === '') progress = `tied`;
+			return Utils.html`<p>Game ${index + 1}: <a href="/${battle.roomid}">${battle.title} - ${progress}</a></p>`;
+		}).join('');
+		if (this.winner) {
+			buf += Utils.html`<p>${this.name(this.winner)} won!</p>`;
+		} else if (this.winner === '') {
+			buf += `<p>The battle was tied.</p>`;
+		}
+		this.room.add(`|controlshtml|<center>${buf}</center>`);
+		this.room.update();
+	}
+
+	startTimer() {
+		this.needsTimer = true;
+		for (const {battle} of this.games) {
+			battle.battle?.timer.start();
+		}
+	}
+
+	onBattleWin(room: Room, winnerid: string) {
+		const loser = this.p1 === winnerid ? this.p2 : this.p1;
+		const loserPlayer = room.battle!.playerTable[loser];
+		if (loserPlayer.hitDisconnectLimit) { // disconnection means opp wins the set
+			this.room.add(`${this.name(loser)} lost by timing out.`);
```

Comment:
Would prefer to say "Player lost the series due to inactivity". We use "due to inactivity" elsewhere and that makes it clear it's the whole best of whatever.